### PR TITLE
Add new Bundler/GemComment cop

### PR DESCRIPTION
**Pull request by sunny**
This cop helps our team make sure all `gem` declarations inside a `Gemfile` have a comment that describes why this gem was added to the project.

Bad:
```rb
gem 'rubocop'
```

Good:
```rb
# Ruby styleguide enforcer.
gem 'rubocop'
```

The cop is disabled by default so as not to break in codebases that do not enforce this behavior.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/bundler/gem_comment.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# TODO: when finished, run `rake generate_cops_documentation` to update the docs
```
Comment (koic) [inline lib/rubocop/cop/bundler/gem_comment.rb]: Please remove this comment.

**Comment by sunny** [comment]
Yay! Thank you for reviewing and merging this! \o/

**Comment by jfelchner** [comment]
@sunny how difficult would it be to add an option to this cop for an inline comment versus a comment above the line?

Example:

```ruby
# This is my gem description
gem 'foo'
```

versus

```ruby
gem 'foo' # This is my gem description
```

When you have a Gemfile that has a long list of gems, adding the comment above can end up making the Gemfile both extremely long, as well as more difficult to parse (inline comments can be lined up so as to be easy to scan).

**Comment by rhymes** [comment]
Does this take into account inline comments?

**Comment by rhymes** [comment]
We use inline comments by copying the first line of the description from rubygems.org.

Example:

```ruby
gem 'oj', '3.6.8'               # The fastest JSON parser and object serializer.
gem 'patron', '0.13.1'          # Ruby HTTP client library based on libcurl
gem 'pg', '1.1.3'               # Pg is the Ruby interface to the PostgreSQL RDBMS
```

**Comment by mikegee** [comment]
@rhymes that's pretty cool 😎 

I prefer to leave comments addressing why the gem was installed in this project and why we're on a particular branch or version, if applicable. Basically, I try to avoid comments that are duplicative with the gem's readme.

For instance:

```ruby
# Get results from the FooBar API over HTTP
# 0.14 has a bug (http://github.com/patron/patron/issue/42)
gem 'patron', '0.13.1'
```

**Comment by rhymes** [comment]
@mikegee that makes perfect sense!

**Pull request merged**